sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/http/ReactorNettyClient.java

```diff
@@ -16,128 +21,46 @@
+        return createWarmedUpReactorNettyClient(httpClientConfig, httpClient);
     }
 
     /**
      * Creates ReactorNettyClient with {@link ConnectionProvider}.
      */
     public static ReactorNettyClient createWithConnectionProvider(ConnectionProvider connectionProvider, HttpClientConfig httpClientConfig) {
-        ReactorNettyClient reactorNettyClient = new ReactorNettyClient();
-        reactorNettyClient.connectionProvider = connectionProvider;
-        reactorNettyClient.httpClientConfig = httpClientConfig;
-        reactorNettyClient.httpClient = reactor.netty.http.client.HttpClient
-            .create(connectionProvider)
-            .observe(getConnectionObserver())
-            .resolver(DefaultAddressResolverGroup.INSTANCE);
-        reactorNettyClient.configureChannelPipelineHandlers();
-        attemptToWarmupHttpClient(reactorNettyClient);
-        return reactorNettyClient;
+        reactor.netty.http.client.HttpClient httpClient = HttpClientUtils.createHttpClient(httpClientConfig, connectionProvider);
+        return createWarmedUpReactorNettyClient(httpClientConfig, httpClient);
     }
 
-    /*
-     * This enables fast warm up of HttpClient
-     */
-    private static void attemptToWarmupHttpClient(ReactorNettyClient reactorNettyClient) {
-        // Warmup wasn't found, so don't attempt it.
-        if (HTTP_CLIENT_WARMUP == null) {
-            return;
-        }
-
-        try {
-            ((Mono<?>) HTTP_CLIENT_WARMUP.invoke(reactorNettyClient.httpClient)).block();
-        } catch (ClassCastException | WrongMethodTypeException throwable) {
-            // Invocation failed.
-            logger.debug("Invoking HttpClient.warmup failed.", throwable);
-        } catch (Throwable throwable) {
-            // Warmup failed.
-            throw new RuntimeException(throwable);
-        }
-    }
-
-    private void configureChannelPipelineHandlers() {
-        Configs configs = this.httpClientConfig.getConfigs();
-
-        if (this.httpClientConfig.getProxy() != null) {
-            this.httpClient = this.httpClient.proxy(typeSpec -> typeSpec.type(ProxyProvider.Proxy.HTTP)
-                .address(this.httpClientConfig.getProxy().getAddress()));
-        }
-
-        if (LoggerFactory.getLogger(REACTOR_NETWORK_LOG_CATEGORY).isTraceEnabled()) {
-            this.httpClient = this.httpClient.wiretap(REACTOR_NETWORK_LOG_CATEGORY, LogLevel.INFO);
-        }
-
-        this.httpClient.secure(sslContextSpec -> sslContextSpec.sslContext(configs.getSslContext()))
-            .option(ChannelOption.CONNECT_TIMEOUT_MILLIS, (int) configs.getConnectionAcquireTimeout().toMillis())
-            .httpResponseDecoder(httpResponseDecoderSpec ->
-                httpResponseDecoderSpec.maxInitialLineLength(configs.getMaxHttpInitialLineLength())
-                    .maxHeaderSize(configs.getMaxHttpHeaderSize())
-                    .maxChunkSize(configs.getMaxHttpChunkSize())
-                    .validateHeaders(true));
```

Comment:
The configurations were not assigned back to this.httpClient at L134, hence lost maxHttpHeaderSize configuration.